File: hashTable/1657_DetermineIfTwoStringsAreClose.java
class Solution {
    public boolean closeStrings(String word1, String word2) {
        if (word1.length() != word2.length()) return false;
        Map<Character, Integer> count1 = new HashMap<>();
        for (int i = 0; i < word1.length(); i++) {
            char c = word1.charAt(i);
            count1.put(c, count1.getOrDefault(c, 0) + 1);
        }
        
        Map<Character, Integer> count2 = new HashMap<>();
        for (int i = 0; i < word2.length(); i++) {
            char c = word2.charAt(i);
            count2.put(c, count2.getOrDefault(c, 0) + 1);
        }
        
        if (!count1.keySet().equals(count2.keySet())) {
            return false;   
        }
        
        List<Integer> sortedCount1 = new ArrayList<>(count1.values());
        Collections.sort(sortedCount1);
        List<Integer> sortedCount2 = new ArrayList<>(count2.values());
        Collections.sort(sortedCount2);
        
        return sortedCount1.equals(sortedCount2);
    }
}

File: hashTable/1657_DetermineIfTwoStringsAreClose.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        # Quick check for length equality
        if len(word1) != len(word2):
            return False
        
        # Use Counter
        count1 = Counter(word1)
        count2 = Counter(word2)
        
        # Compare the characters directly
        return count1.keys() == count2.keys() and sorted(count1.values()) == sorted(count2.values())
        